src/modules/hip/kernel/erase.hpp

```diff
@@ -0,0 +1,278 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - Erase main kernels --------------------
+template <typename T, typename U>
+__global__ void erase_pkd_hip_tensor(T *dstPtr,
+                                     uint2 dstStridesNH,
+                                     RpptRoiLtrb *anchorBoxInfoTensor,
+                                     U *colorsTensor,
+                                     Rpp32u *numBoxesTensor,
+                                     RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        return;
+
+    Rpp32u numBoxes = numBoxesTensor[id_z];
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    // check if the co-ordinates is within any user defined box
+    for (int i = 0; i < numBoxes; i++)
+    {
+        int temp = (id_z * numBoxes) + i;
+        if (id_x >= anchorBoxInfoTensor[temp].lt.x && id_x <= anchorBoxInfoTensor[temp].rb.x && id_y >= anchorBoxInfoTensor[temp].lt.y && id_y <= anchorBoxInfoTensor[temp].rb.y)
+        {
+            *reinterpret_cast<U *>(dstPtr + dstIdx) = static_cast<U>(colorsTensor[temp]);
+            break;
```

Comment:
why do you break here? how does it handle all boxes?

Reply:
If a particular (x, y) is "erased" to say Blue color, it kindof also means that the same (x, y) can't be erased again, so we don't need to check for the rest of the for loop.

https://github.com/ROCm/rpp/pull/336/commits/56d368bc29e01ac81e2c57026acf4ea62dd65682 added for clarity from a user input standpoint.